Source/EventFlow/Sagas/SagaErrorHandler.cs

```diff
@@ -29,14 +29,29 @@ namespace EventFlow.Sagas
 {
     public class SagaErrorHandler : ISagaErrorHandler
     {
+        private readonly Func<Type, ISagaErrorHandler> _sagaErrorHandlerFactory;
+
+        public SagaErrorHandler(Func<Type, ISagaErrorHandler> sagaErrorHandlerFactory)
+        {
+            _sagaErrorHandlerFactory = sagaErrorHandlerFactory;
+        }
+
         public Task<bool> HandleAsync(
             ISagaId sagaId,
             SagaDetails sagaDetails,
             Exception exception,
             CancellationToken cancellationToken)
         {
-            // The default handler cannot handle anything!
+            // Search for a specific SagaErrorHandler<Saga> based on saga type
+            ISagaErrorHandler sagaErrorHandler = _sagaErrorHandlerFactory(sagaDetails.SagaType);
```

Comment:
@ProH4Ck I think this should be moved to the `DispatchToSagas` instead. Doing it here makes it harder to do a "catch all" error handler as well as you will need to copy the code to handle the specific error handlers yourselv.

Reply:
Correct. I'll push changes to the PR in the evening